### PR TITLE
.github: split integration tests CI into jobs

### LINKED ISSUE #5440: Split CI for integration testing and code quality
### Description and goal

The job for integration tests is getting crowded: flake8, pytype, pytest, etc.
While they share some dependencies, splitting may help with readability and performance.

See https://github.com/osrd-project/osrd/pull/5419

### Acceptance criteria

We should have a testing job and a code quality job (similar to editoast)

### PR DESCRIPTION
**Pull request by shenriotpro**
closes https://github.com/osrd-project/osrd/issues/5440

### CODE REVIEW

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -11,8 +13,8 @@ on:
       - prod
 
 jobs:
-  integration_tests:
-    name: Integration tests
+  integration_tests_quality:
```
Comment (flomonster) [inline .github/workflows/integration_tests.yml]: This job is run for all PR. Ideally, it should be triggered only when changes are made in the `/tests` folder. The easy way to do so is to have another workflow. Maybe it's possible to configure it in a job.

**Review by flomonster — CHANGES_REQUESTED** [review]
Thanks for the PR!

We need to fix the trigger condition otherwise there is no point in splitting the job in two.

**Comment by shenriotpro** [comment]
> Thanks for the PR!
> 
> We need to fix the trigger condition otherwise there is no point in splitting the job in two.

It's still nice to have both jobs run in parallel, but I'll take a look.

File: .github/workflows/integration_tests_quality.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          cd tests
+          poetry run black . --check
+      - name: Isort
+        run: |
+          cd tests
+          poetry run isort . --check
+      - name: Pytype
+        run: |
+          cd tests
+          poetry run pytype -j auto
```
Comment (Khoyo) [inline .github/workflows/integration_tests_quality.yml]: Missing newline at EOF

**Review by Khoyo — APPROVED** [review]

**Review by flomonster — APPROVED** [review]
LGTM!

**Pull request merged**